### DISCUSSION TITLE
N64 Dumps not complete

### DISCUSSION THREAD

**Discussion by Ibralosh** [sanni/cartreader#150] [Need help]
I tried to dump several N64 carts and it seems that my cartreader is dumping these carts only partly. 0x00000000 to 0x000001ff is written with 0x0000, 0x00000200 to 0x000003FF with 0x0200, 0x00000400 to 0x000005FF with 0x0400, ... .

0x00001000 to 0x000017FF is showing correct data, but 0x00001800 to 0x00002FFF is showing again the pattern that is mentioned above (starting with 0x1800). 0x0003000 to 0x000037FF is then showing correct data again. And so on...

I took a first look into the code, but I didn't find a possible cause for that. Do you have any ideas?

**Comment by sanni** [comment]
Might be a bad solder connection or a short circuit between two neighboring pins.

Download the [pinout.xls](https://github.com/sanni/cartreader/blob/master/pinout.xls?raw=true) and measure the resistance between each pin on the cartridge slot(column N64 rom) and the corresponding pin on the backside of the Arduino Mega(column Arduino). 
The resistance of each connection should be below 1 ohm. If one pin is higher, re-solder that connection.

Next make sure that no AD# pin on the N64 slot is connected to another AD# pin.

**Reply by Ibralosh** [reply]
I doublechecked that. The connections to the Arduino Mega are OK and there are no shorts between AD# pins. I guess I will have to check if my Arduino is damaged.

**Comment by sanni** [comment]
I get this pattern when I remove [line 499 from N64.ino](https://github.com/sanni/cartreader/blob/master/Cart_Reader/N64.ino#L499), so check the ale_h and ale_L pins, your error probably is there. 

![afwsfaaaedawa](https://user-images.githubusercontent.com/19169494/118488180-c4483a80-b71b-11eb-922a-e2dcb05d2581.jpg)

**Reply by Ibralosh** [reply]
Thanks a lot for this hint! I gave these signals a closer look today and I found that inside the Game Boy connector pins 22 and 23 were a little bent and touched each other. This connection also connected aleH to aleL.
Disconnecting these pins did the trick, now N64 dumps are working.
Thank you so much!
